docs/intents/getQuote.md

```diff
@@ -0,0 +1,363 @@
+
+  // Desired amount to receive on target chain (as integer string in wei)
+  targetAmount: string
+
+  // Destination blockchain ID where user wants to receive tokens
+  targetChain: chainId
+
+  // Destination token contract address (hex string, e.g., "0x...")
+  targetTokenAddress: string
+
+  // Optional: specify which routing algorithm to use
+  algorithm?: 'greedy' | 'dynamic-programming' | 'dijkstra' | 'multi-objective' | 'genetic' | 'simulated-annealing' | 'ant-colony' | 'a-star'
+
+  // Optional user preferences for route optimization
+  userPreferences?: {
+    // Whether to prioritize speed over cost (boolean)
+    prioritizeSpeed: boolean
+
+    // Maximum acceptable slippage (percentage, e.g., 0.5 = 0.5%)
+    maxSlippage: number
+
+    // Weight for cost optimization (0-1, where 1 = cost is most important)
+    costWeight: number
+  }
+}
+
+```
+
+### **Output Schema:**
+
+```tsx
+interface ServerSideQuoteResponse {
+  route: {
+    // Array of bridge transfers needed to complete the route
+    steps: Array<{
+      // Source blockchain ID for this step
+      sourceChain: chainId
+      
+      // Source token address for this step
+      sourceTokenAddress: Hex
+
+      // Amount to bridge from this chain (as integer string in wei)
+      amount: string
+
+      // Bridge fee for this step (in wei of source chain's native token)
+      fee: string
+
+      // Estimated gas cost (in gas units, not wei)
+      gasEstimate: string
+      
+      // Estimated gas cost in USD (in USD)
+      usdGasEstimate: string
+
+      // Estimated completion time for this step (in seconds)
+      timeEstimate: number
+
+      // Name/identifier of the bridge provider (string)
+      bridgeProvider: string
+    }>
+
+    // Sum of all fees across all steps (in wei of respective native tokens)
+    totalFee: string
+
+    // Sum of all gas estimates (in gas units)
+    totalGasEstimate: string
+    
+    // Sum of all USD gas estimates (decimal representation in USD)
+    usdTotalGasEstimate: string
+
+    // Estimated time for entire route completion (in seconds)
+    estimatedCompletionTime: number
+  }
+
+  // Algorithm used to calculate this route
+  algorithmUsed: string
+
+  // Unique identifier for this quote (for tracking/execution)
+  quoteId: string
+
+  // Unix timestamp when this quote expires
+  validUntil: timestamp
+}
+
+```
+
+### **Advantages:**
+
+- **Simplified Integration**: Wallets receive ready-to-execute instructions
+- **Optimized Performance**: Server can run complex algorithms without client constraints
+- **Reduced Bandwidth**: Minimal data transfer for simple use cases
+- **Centralized Logic**: Route optimization improvements benefit all clients immediately
+- **Algorithm Flexibility**: Different algorithms can be tested and compared server-side
+
+### **Disadvantages:**
+
+- **Limited Transparency**: Users cannot see alternative options or understand trade-offs
```

Comment:
We can potentially change the response format to be an array of options, each option being the currently proposed schema, but allowing solvers to return different options that the client then can decide to show to the users

Reply:
That is a good idea.